core/src/main/java/io/questdb/cairo/wal/seq/TableSequencerAPI.java

```diff
@@ -0,0 +1,433 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.wal.seq;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoEngine;
+import io.questdb.cairo.CairoError;
+import io.questdb.cairo.TableStructure;
+import io.questdb.cairo.pool.AbstractPool;
+import io.questdb.cairo.pool.PoolListener;
+import io.questdb.cairo.pool.ex.PoolClosedException;
+import io.questdb.cairo.wal.WalWriter;
+import io.questdb.griffin.engine.ops.AlterOperation;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.*;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.StringSink;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayDeque;
+import java.util.Iterator;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static io.questdb.cairo.wal.WalUtils.SEQ_DIR;
+
+public class TableSequencerAPI extends AbstractPool {
+    private static final Log LOG = LogFactory.getLog(TableSequencerAPI.class);
+    private final ConcurrentHashMap<TableSequencerEntry> seqRegistry = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<WalWriterPool> walRegistry = new ConcurrentHashMap<>();
+    private final CairoEngine engine;
+
+    public TableSequencerAPI(CairoEngine engine, CairoConfiguration configuration) {
+        super(configuration, configuration.getInactiveWriterTTL()); //todo: separate config option
```

Comment:
nit: probably it's worth it to address this TODO or remove it.

Reply:
added the config